les/vflux/server/balance.go

```diff
@@ -546,23 +564,23 @@ func (n *NodeBalance) balanceToPriority(b balance, capacity uint64) int64 {
 }
 
 // reducedBalance estimates the reduced balance at a given time in the fututre based
-// on the current balance, the time factor and an estimated average request cost per time ratio
-func (n *NodeBalance) reducedBalance(at mclock.AbsTime, capacity uint64, avgReqCost float64) balance {
-	dt := float64(at - n.lastUpdate)
-	b := n.balance
+// on the given balance, the time factor and an estimated average request cost per time ratio
+func (n *NodeBalance) reducedBalance(b balance, start mclock.AbsTime, dt time.Duration, capacity uint64, avgReqCost float64) balance {
+	at := start + mclock.AbsTime(dt/2)
```

Comment:
why it's `start + diff/2`?

Reply:
Because it estimates a reduction of balance applied continuously during the `dt` time period and expiration during that time so the expiration factor applied would theoretically also vary between `start` and `start+dt`. The simplest and closest way to approximate this is to just use an exp factor calculated at `start+dt/2`. It makes minimal difference but it's also not a lot of complexity, maybe I'll add a comment about it.

Reply:
Yes.....

Reply:
Although I don't like the estimation. But our balance token is dynamic which already introduces a lot of complexity. It's ok to do this compromise here.